langchain/retrievers/document_compressors/chain_extract.py

```diff
@@ -69,9 +69,10 @@ def from_llm(
         llm: BaseLanguageModel,
         prompt: Optional[PromptTemplate] = None,
         get_input: Optional[Callable[[str, Document], str]] = None,
+        **llm_chain_kwargs: dict[str, Any]
```

Comment:
From my understanding, this is for llm_chain creation, but the purpose of this function is to load from llm directly with no additional customization of llm_chain.

If my understanding is correct, I would suggest the following
1. construct a LLMChain outside and then create LLMChainExtractor.
2. if you need further convenience beyond step 1, I would suggest writing a static utility function with your preferred input and output in this file or another file in this folder.

I could be wrong. Will wait for maintainer to further comment here. :)

Reply:
👋 @skcoirz see context in https://github.com/hwchase17/langchain/issues/3747, this is the same as in `BaseQAWithSourcesChain` https://github.com/hwchase17/langchain/blob/72c5c15f7fdc1918880e3cfd0949199e5a0b5bda/langchain/chains/qa_with_sources/base.py#L40-L47

Reply:
Yeah, got you! Thank you for sharing the example! This is just my personal idea, please let me know if this doesn’t make sense to you. :)

In the example you shared, the `kwargs` is passed to the class itself. So we can consider this function as an entry to initiate this class itself. Code below:
https://github.com/hwchase17/langchain/blob/72c5c15f7fdc1918880e3cfd0949199e5a0b5bda/langchain/chains/qa_with_sources/base.py#L39-L64

But in the PR here, the parameter `kwargs` is used as input for LLMChain which is a intermediate layer between the function and the class. From my understanding, a more suitable solution is to build another entry function which build this extractor from LLMChain directly, which, however, is overlapping with the default creation method of this class. That’s why I think it’s easier to create a LLMChain outside and then build this class directly.

Please let me know if this makes sense to you. Thank you!

Reply:
Sure, that sounds good, but then I/user would need to initialize the LLMChain much the same way it's being done in the `from_llm` (which is a couple of extra lines of code, just to pass `verbose=True`). I really appreciate the convenience of the `from_llm` method (where I can reuse existing LLM object), so why not (in my opinion) improve it slightly and let the user pass kwargs to the internal `LLMChain`.

Reply:
Ah that makes sense. If verbose is the only param which we need, maybe build a another wrapper function on top of from_LLM with name postfix _with_verbose ? In this way, we can get the verbose and we don't need to worry about other parameters in kwargs. How do you think?

Reply:
@skcoirz `verbose` is just an example, a specialized `_with_verbose` seems a bit smelly.

Reply:
Sure, it's just my 2 cents. :)

Reply:
Sounds good, appreciate that :)